x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/text_expansion_callout.tsx

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { useValues } from 'kea';
+
+import {
+  EuiBadge,
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiPanel,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage, FormattedHTMLMessage } from '@kbn/i18n-react';
+
+import { docLinks } from '../../../../../shared/doc_links';
+
+import { MLInferenceLogic } from './ml_inference_logic';
+
+export interface TextExpansionCallOutState {
+  dismiss: () => void;
+  dismissable: boolean;
+  show: boolean;
+}
+
+export interface TextExpansionCallOutProps {
+  dismissable?: boolean;
+}
+
+export const TEXT_EXPANSION_CALL_OUT_DISMISSED_KEY =
+  'enterprise-search-text-expansion-callout-dismissed';
+
+export const useTextExpansionCallOutData = ({
+  dismissable = false,
+}: TextExpansionCallOutProps): TextExpansionCallOutState => {
+  const { supportedMLModels } = useValues(MLInferenceLogic);
+
+  const doesNotHaveTextExpansionModel = useMemo(() => {
+    return !supportedMLModels.some((m) => m.inference_config?.text_expansion);
+  }, [supportedMLModels]);
+
+  const [show, setShow] = useState<boolean>(() => {
+    if (!dismissable) return true;
+
+    try {
+      return localStorage.getItem(TEXT_EXPANSION_CALL_OUT_DISMISSED_KEY) !== 'true';
+    } catch {
+      return true;
+    }
+  });
+
+  useEffect(() => {
+    try {
+      localStorage.setItem(TEXT_EXPANSION_CALL_OUT_DISMISSED_KEY, JSON.stringify(!show));
+    } catch {
+      return;
+    }
+  }, [show]);
+
+  const dismiss = useCallback(() => {
+    setShow(false);
+  }, []);
+
+  return { dismiss, dismissable, show: doesNotHaveTextExpansionModel && show };
+};
+
+export const TextExpansionCallOut: React.FC<TextExpansionCallOutProps> = (props) => {
+  const { dismiss, dismissable, show } = useTextExpansionCallOutData(props);
+
+  if (!show) return null;
+
+  return (
+    <EuiPanel color="success">
+      <EuiFlexGroup direction="column" gutterSize="s">
+        <EuiFlexGroup direction="row" gutterSize="s" alignItems="center">
+          <EuiFlexItem grow={false}>
+            <EuiBadge color="success">
+              <FormattedMessage
+                id="xpack.enterpriseSearch.content.index.pipelines.textExpansionCallOut.titleBadge"
+                defaultMessage="New"
+              />
+            </EuiBadge>
+          </EuiFlexItem>
+          <EuiFlexItem grow>
+            <EuiTitle size="xs">
+              <h4>
+                <EuiText color="text">
+                  <FormattedMessage
```

Comment:
@sloanelybutsurely Looking at the [latest design](https://www.figma.com/file/JFJIID1ufVi7ZgOtZYk7zQ/SLIM-Pipeline-UI?node-id=81-128301&t=gAnfXV1U11Mx42MG-0) this should be bold text - can you please verify?

Reply:
🔧 fixed: https://github.com/elastic/kibana/pull/154603/commits/7755e9a59fc74eed6f2ebb695c3329f5f7eded4f

![Screen Shot 2023-04-11 at 09 24 03](https://user-images.githubusercontent.com/1699281/231176790-2f9aaa19-2a8b-42a5-bf3b-581832fbcc17.png)